src/library/Collections.nim

```diff
@@ -615,15 +615,26 @@ proc defineLibrary*() =
             if checkAttr("n"):
                 if xKind == String:
                     if x.s.len == 0: push(newString(""))
-                    else: push(newString(x.s[0..aN.i-1]))
+                    else:
+                        if aN.i > x.s.len:
+                            push(newString(x.s[0..^1]))
+                        else: 
+                            push(newString(x.s[0..aN.i-1]))
```

Comment:
I'm seeing some CI build fails, so I'm not sure if I'm supposed to... "review" this one, but I go ahead and do it either way lol 🤣 

The code looks good, although I'm wondering if we could use e.g. [`min`](https://nim-lang.org/docs/system.html#min%2Cint%2Cint), instead of these different `if`-`else`s (performance-wise it should make a tiny bit of a difference, since `min` is a Nim "magic" proc, plus it could look cleaner... just saying 😉 )

Like: 
```suggestion
                        push(newString(x.s[0..min(aN.i,x.s.len)-1]))
```

What do you think?